lang/types/value.go

```diff
@@ -593,14 +669,26 @@ func (obj *ListValue) Copy() Value {
 
 // Value returns the raw value of this type.
 func (obj *ListValue) Value() interface{} {
-	typ := obj.T.Reflect()
-	// create an empty slice (of len=0) with room for cap=len(obj.V) elements
-	val := reflect.MakeSlice(typ, 0, len(obj.V))
+	return interfaceOf(obj)
+}
 
-	for _, x := range obj.V {
-		val = reflect.Append(val, reflect.ValueOf(x.Value())) // recurse
+// Into sets the value of the given reflect.Value to the value of this obj
+func (obj *ListValue) Into(val reflect.Value) {
+	val = FollowPtr(val)
+	count := len(obj.V)
+	if count > val.Cap() {
+		// TODO: should this just use len(obj.V) for cap
```

Comment:
Want a second opinion here. Is this premature optimisation? Should I just drop the whole power-of-two thing and just use `len(obj.V)` instead? Is the additional memory allocation wasteful?

Reply:
Decided to keep this after a discussion with James. We're mostly indifferent, but this more closely resembles what the Go runtime does internally